### DISCUSSION TITLE
Is it possible to Build NetBeans 13

### DISCUSSION THREAD

**Discussion by jmccay-work** [apache/netbeans#5218] [Q&A] [answered]
Is it possible to build NetBeans 13 with the following:

Java version:
- openjdk version "1.8.0_292"
- OpenJDK Runtime Environment (build 1.8.0_292-b10)
- OpenJDK Server VM (build 25.292-b10, mixed mode)

Gradle version:
- Gradle: 6.8.2
- Kotlin: 1.4.20
- Groovy: 2.5.12
- Ant: Apache Ant(TM) version 1.10.9 compiled on September 27 2020

I would like to build NetBeans 13 to use a prepackaged Gradle 6.8.2 and Java 8 (32) by default.  If it is possible, what flags do I have to set?  What would I need to change or do to complete it?

**Comment by mbien** [comment, marked as answer]
i don't know if it was still possible to build NB 13 on JDK 8, but you could give it a try and see how much of it is buildable. JDK 11 is the current requirement for NB 16.

Simply check out the github tag for 13 and follow the usual build instructions:
https://github.com/apache/netbeans#building-netbeans

> I would like to build NetBeans 13 to use a prepackaged Gradle 6.8.2 and Java 8 (32) by default. 

NetBeans does not package a JDK. There are community packages available though which do package a JDK via installers for example. They are linked from the download page.

> Java 8 (32) 

i suppose this stands for 32bit? We don't really test on 32bit JVMs since most vendors discontinued them and the main usecases are gone since the introduction of compressed oops on 64bit JVMs. It may or may not work.

NB 16 can also use other gradle/maven versions then those which are bundled with it by default you sure your endeavor is actually needed? Some bugfixes were recently deployed via [16-u1](https://github.com/apache/netbeans/releases/tag/16-u1)

**Reply by neilcsmith-net** [reply]
Officially, NB 13 was the release we required building on JDK 11.  It's possible we didn't actually break it on JDK 8 until NB 14 though.
